src/test/java/org/cloudfoundry/promregator/endpoint/DiscoveryEndpointControllerTest.kt

```diff
@@ -0,0 +1,65 @@
+package org.cloudfoundry.promregator.endpoint
+
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.every
+import org.cloudfoundry.promregator.config.PromregatorConfiguration
+import org.cloudfoundry.promregator.discovery.CFMultiDiscoverer
+import org.cloudfoundry.promregator.scanner.Instance
+import org.cloudfoundry.promregator.scanner.ResolvedTarget
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+
+@ExtendWith(SpringExtension::class)
+@WebMvcTest(DiscoveryEndpoint::class, properties = ["promregator.simulation.enabled=true", "promregator.lifecycle.enabled=false"])
+class DiscoveryEndpointControllerTest(
+       @Autowired private val mockMvc: MockMvc
```

Comment:
Here is a Kotlin overview to get you up to speed (and to highlight some of the benefits):

1 - You can define properties in your constructor (`private val` is essentially creating a field in the test for `mockMvc`, and flagging it as final so it can't be changed. It also creates getters/setters that Java code could call, if it weren't `private`. The `val`/`var` support in Kotlin allows the compiler to do additional checks against type and nullability as well (aka 'smart casts').

2 - Constructor injection is supported in JUnit 5 tests (which lets Spring dependency injection work, and still make them `val`s